File: C++ Code/bubblesort.cpp
#include <iostream>
#include <chrono>
#include <fstream>

using namespace std;
using namespace chrono;

void bubbleSort(int arr[], int size) {

    for (int i = 1; i <= size - 1; i++) {

        for (int j = 0; j < size - i; j++) {

            if (arr[j] > arr[j + 1]) {

                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;

            }

        }

    }

}
int main() {

     ofstream file(/*file name*/);

   
    file << "Run,Time (nanoseconds)" << endl;

    for (int i = 1; i <= 10; i++){
        int arr[] = {/*Array*/  };

        int size = sizeof(arr) / sizeof(arr[0]);

        // Start timer
        auto start = high_resolution_clock::now();

        // Sort array
        bubbleSort(arr, size);

        // Stop timer
        auto stop = high_resolution_clock::now();

        // Calculate execution time
        auto duration = duration_cast<nanoseconds>(stop - start);

        cout << "Bubble Sort Execution Time: " << duration.count() << " nanoseconds" << endl;

        file << i << "," << duration.count() << endl;
    
    }

    file.close();

    return 0;
}

File: Julia Code/bubblesort.jl
function bubbleSort!(arr, size)

    for i in 1:(size - 1)

        for j in 1:(size - i)

            if arr[j] > arr[j + 1]

                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp

            end

        end

    end

end

function main()
    file = open(#=file name=#, "w")
     println(file, "Run,Time (nanoseconds)")
    for i in 1:100

        # Full original array goes here unchanged
        
        arr = #array
        size = length(arr)

        # Start timer
        start = time_ns()

        # Sort array
        bubbleSort!(arr, size)

        # Stop timer
        stop = time_ns()

        # Calculate execution time
        duration = stop - start

       println("Bubble Sort Execution Time: ", duration, " nanoseconds")
       println(file, i, ",", duration)

    end

    close(file)
end
main()
